File: examples/python/basic.py
"""
Invoice Generator API - Basic Usage Example

This example demonstrates the basic usage of the Invoice Generator API.
API Documentation: https://docs.apiverve.com/ref/invoicegenerator
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/invoicegenerator'

def call_invoicegenerator_api():
    """
    Make a POST request to the Invoice Generator API
    """
    try:
        # Request body
        request_body &#x3D; {
    &#x27;invoiceNumber&#x27;: &#x27;INV000001&#x27;,
    &#x27;date&#x27;: &#x27;2025-02-01&#x27;,
    &#x27;dueDate&#x27;: &#x27;2025-11-30&#x27;,
    &#x27;from_name&#x27;: &#x27;John Doe&#x27;,
    &#x27;from_street&#x27;: &#x27;123 Elm St&#x27;,
    &#x27;from_city&#x27;: &#x27;Springfield&#x27;,
    &#x27;from_state&#x27;: &#x27;IL&#x27;,
    &#x27;from_zip&#x27;: &#x27;62701&#x27;,
    &#x27;to_name&#x27;: &#x27;Jane Smith&#x27;,
    &#x27;to_street&#x27;: &#x27;456 Oak St&#x27;,
    &#x27;to_city&#x27;: &#x27;Springfield&#x27;,
    &#x27;to_state&#x27;: &#x27;IL&#x27;,
    &#x27;to_zip&#x27;: &#x27;62702&#x27;,
    &#x27;job&#x27;: &#x27;Web Development&#x27;,
    &#x27;paymentTerms&#x27;: &#x27;Net 30&#x27;,
    &#x27;discount&#x27;: 10,
    &#x27;salesTax&#x27;: 37.07,
    &#x27;currency&#x27;: &#x27;USD&#x27;,
    &#x27;items&#x27;: [
        {
            &#x27;qty&#x27;: 2,
            &#x27;description&#x27;: &#x27;Web Design Services&#x27;,
            &#x27;unit_price&#x27;: 500
        },
        {
            &#x27;qty&#x27;: 1,
            &#x27;description&#x27;: &#x27;Domain Registration&#x27;,
            &#x27;unit_price&#x27;: 100
        }
    ]
}

        headers = {
            'x-api-key': API_KEY,
            'Content-Type': 'application/json'
        }

        response = requests.post(API_URL, headers=headers, json=request_body)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Invoice Generator API...\n')

    result = call_invoicegenerator_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: examples/javascript/basic.js
/**
 * Invoice Generator API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the Invoice Generator API.
 * API Documentation: https://docs.apiverve.com/ref/invoicegenerator
 */

const API_KEY = process.env.APIVERVE_API_KEY || 'YOUR_API_KEY_HERE';
const API_URL = 'https://api.apiverve.com/v1/invoicegenerator';

/**
 * Make a POST request to the Invoice Generator API
 */
async function callInvoiceGeneratorAPI() {
  try {
    // Request body
    const requestBody &#x3D; {
    &quot;invoiceNumber&quot;: &quot;INV000001&quot;,
    &quot;date&quot;: &quot;2025-02-01&quot;,
    &quot;dueDate&quot;: &quot;2025-11-30&quot;,
    &quot;from_name&quot;: &quot;John Doe&quot;,
    &quot;from_street&quot;: &quot;123 Elm St&quot;,
    &quot;from_city&quot;: &quot;Springfield&quot;,
    &quot;from_state&quot;: &quot;IL&quot;,
    &quot;from_zip&quot;: &quot;62701&quot;,
    &quot;to_name&quot;: &quot;Jane Smith&quot;,
    &quot;to_street&quot;: &quot;456 Oak St&quot;,
    &quot;to_city&quot;: &quot;Springfield&quot;,
    &quot;to_state&quot;: &quot;IL&quot;,
    &quot;to_zip&quot;: &quot;62702&quot;,
    &quot;job&quot;: &quot;Web Development&quot;,
    &quot;paymentTerms&quot;: &quot;Net 30&quot;,
    &quot;discount&quot;: 10,
    &quot;salesTax&quot;: 37.07,
    &quot;currency&quot;: &quot;USD&quot;,
    &quot;items&quot;: [
        {
            &quot;qty&quot;: 2,
            &quot;description&quot;: &quot;Web Design Services&quot;,
            &quot;unit_price&quot;: 500
        },
        {
            &quot;qty&quot;: 1,
            &quot;description&quot;: &quot;Domain Registration&quot;,
            &quot;unit_price&quot;: 100
        }
    ]
};

    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'x-api-key': API_KEY,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(requestBody)
    });

    // Check if response is successful
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();

    // Check API response status
    if (data.status === 'ok') {
      console.log('✓ Success!');
      console.log('Response data:', data.data);
      return data.data;
    } else {
      console.error('✗ API Error:', data.error || 'Unknown error');
      return null;
    }

  } catch (error) {
    console.error('✗ Request failed:', error.message);
    return null;
  }
}

// Run the example
callInvoiceGeneratorAPI()
  .then(result => {
    if (result) {
      console.log('\n📊 Final Result:');
      console.log(JSON.stringify(result, null, 2));
    }
  });
